scripts/ghprb_deduper.py

```diff
@@ -0,0 +1,237 @@
+                a list of 'builds' for the PR. Each item in the list
+                has the data for the build as returned in self.get_json.
+        """
+        build_data = defaultdict(list)
+
+        # This is used to find the PR id instead of having a bunch of
+        # messy loops.
+        pr_id_regex = (r"{[\s-]*\"name\"[\s-]*:[\s-]*\"ghprbPullId\""
+                       r"[\s-]*,[\s-]*\"value\"[\s-]*:[\s-]*\"[\d]*\"}")
+
+        for b in data['builds']:
+            if b['building']:
+                pr_id_data_str = re.search(
+                    pr_id_regex, json.dumps(b['actions'])).group(0)
+                pr_id_data = json.loads(pr_id_data_str)
+                pr_id = pr_id_data['value']
+                build_data[pr_id].append(b)
+
+        return build_data
+
+    def stop_duplicates(self, build_data):
+        """
+        Finds PRs that have multiple builds actively running.
+
+        :Args:
+            build_data: the data returned by self.get_running_builds()
+        """
+        lines = []
+        for pr, builds in build_data.iteritems():
+            if len(builds) > 1:
+                # Sort builds by timestamp (newest to oldest)
+                sorted_builds = sorted(
+                    builds, key=itemgetter('timestamp'), reverse=True)
+
+                # Most recently started build is the first in the list
+                current_build_id = sorted_builds[0]['number']
+
+                # Any other builds are assumed to be outdated
+                old_build_ids = [b['number'] for b in sorted_builds[1:]]
+
+                lines.append("PR #{}:".format(pr))
+                lines.append("\tNum running builds: {}".format(
+                    len(sorted_builds)))
+                lines.append("\tCurrent build: {}".format(
+                    current_build_id))
+                lines.append("\tOutdated builds: {}".format(
+                    old_build_ids))
+
+                desc = self._aborted_description(current_build_id, pr)
+
+                for b in old_build_ids:
+                    try:
+                        self.update_build_desc(b, desc)
+                        self.stop_build(b)
+                    except Exception as e:
+                        logger.error(e.message)
+
+        if lines:
+            out = ("\n---------------------------------"
+                   "\n** Extra running builds found. **"
+                   "\n---------------------------------\n")
+            out += "\n".join(lines)
+            logger.info(out)
+        else:
+            logger.info("No extra running builds found.")
+
+    def update_build_desc(self, build_id, description):
+        """
+        Updates build description.
+
+        :Args:
+            build_id: id number of build to update
+            description: the new description
+        """
+        build_url = self._append_url(self.job_url, str(build_id))
+        url = self._append_url(build_url, "/submitDescription")
```

Comment:
We'll need to make sure the user that makes these edits has this level of access.